### PR TITLE
fix(learn): update the return Object.values & solution

### LINKED ISSUE #40444: Bug in add method of  "Perform a Union on Two Sets " challenge seed
**Describe your problem and how to reproduce it:**

The [perform a union on two sets challenge](https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/perform-a-union-on-two-sets) in coding interview prep, data structures section requires one to create a method called `union` on the defined `Set` data structure in the challenge seed. This method should take another Set as an argument and return the union of the two sets, excluding any duplicate values. Essentially none of the solutions I submitted passed the tests. On closer inspection I noticed that the `set` objects used for testing the method are not sets. If you want to see what I mean, go to the challenge and add the following `union` method and click "run the tests".
```js
union(set){
    console.log(set);
    console.log(this.dictionary);
  }
```
You will see the following on the console.
```js
{ dictionary: { c: true, d: true }, length: 2 } // passed in to  union
{ a: true, b: true, c: true } // this.dictionary
```
None of the above are sets otherwise we wouldn't be having `true` appearing more than once in each of the above "sets".
I think this is caused by a bug in `add` method of [the project seed](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/data-structures/perform-a-union-on-two-sets.md#--seed--). Essentially the `add` method in the project seed is:
```js
 add(element) {
    if (!this.has(element)) {
      this.dictionary[element] = true; // This should have been element
      this.length++;
      return true;
    }

    return false;
  }
```
Instead of adding the passed element, `true` is instead being added to the set. 

There also seems to be a contradiction between the instructions and what is actually expected in the test.
### The test instruction says

> The union of ["a", "b", "c"] and ["c", "d"] should return ["a", "b", "c", "d"].

From the example given, it appears an array is being passed to and returned from the union method  yet in actual sense an instance of `Set` is passed to and returned from the `union` method.

This issue of returning `keys` in the `values` method as highlighted in issue #40443 seems to have been carried to this challenge.
```js
values() {
    return Object.keys(this.dictionary);
  }
```

**Add a Link to the page with the problem:**

https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/perform-a-union-on-two-sets

**Tell us about your browser and operating system:**
* Browser Name: N/A
* Browser Version: N/A
* Operating System: N/A

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by qaz6209031**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40444

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/perform-a-union-on-two-sets.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ class Set {
   }
   // This method will return all the values in the set
   values() {
-    return Object.keys(this.dictionary);
+    return Object.values(this.dictionary);
   }
   // This method will add an element to the set
   add(element) {
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/perform-a-union-on-two-sets.md:72]: The `add` method in the challenge seed code will need to be updated to reflect the change as well.

Currently line 74 is still `this.dictionary[element] = true;`.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Hey there,

Thank you for opening this pull request! I have just one quick nitpick. 

Related: https://github.com/freeCodeCamp/freeCodeCamp/pull/40450

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/perform-a-union-on-two-sets.md (hunk 1 above)
Comment (qaz6209031) [reply curriculum/challenges/english/10-coding-interview-prep/data-structures/perform-a-union-on-two-sets.md:72]: Thanks for pointint that out, now I fixed it

**Comment by nhcarrigan** [comment]
In order to comprehensively resolve the linked issue, I think we also need to address this line:

>The union of ["a", "b", "c"] and ["c", "d"] should return ["a", "b", "c", "d"].

It could possibly be changed to 

> The union of a `Set` containing values ["a", "b", "c"] and a `Set` containing values ["c", "d"] should return a new `Set` containing values ["a", "b", "c", "d"].

**Comment by qaz6209031** [comment]
> In order to comprehensively resolve the linked issue, I think we also need to address this line:
> 
> > The union of ["a", "b", "c"] and ["c", "d"] should return ["a", "b", "c", "d"].
> 
> It could possibly be changed to
> 
> > The union of a `Set` containing values ["a", "b", "c"] and a `Set` containing values ["c", "d"] should return a new `Set` containing values ["a", "b", "c", "d"].

**Comment by qaz6209031** [comment]
Yeah that makes more sense, I fixed it in the newest commit

**Review by naomi-lgbt — APPROVED** [review]
LGTM~! 🥳 Appears to resolve the concerns in the linked issue.

Thank you for your pull request! We look forward to additional contributions! Please note that we are in a code freeze, so this cannot be merged until that freeze is lifted.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**